src/register.rs

```diff
@@ -166,9 +165,25 @@ impl DataRate {
     }
 }
 
+/// Data status structure. Decoded from `STATUS_REG` register.
+#[derive(Debug)]
+pub struct DataStatus {
+    /// ZYXOR bit
```

Comment:
Can you explain what these mean a bit better?

Reply:
This was again just taken from the `lis2dh12` library, so I'm not too attached to it. I agree it's not super clear without context.

`STATUS_REG` has the following bit fields:

* `ZYXOR` - X, Y and Z-axis data overrun
* `ZOR` - Z-axis data overrun
* `YOR` - Y-axis data overrun
* `XOR` - X-axis data overrun
* `ZYXDA` - X, Y and Z-axis new data available
* `ZDA` - Z-axis new data available
* `YDA` Y-axis new data available
* `XDA` X-axis new data available

So this struct basically just splits the fields up into groups for convenience:

`zyxor` -> `ZYXOR`  
`xyzor` -> (`XOR`, `YOR`, `ZOR`)  
`zyxda` -> `ZYXDA`  
`xyzda` -> (`XDA`, `YDA`, `ZDA`)  

I have no problem changing it if there's a better way to do this, just seemed reasonable enough to me that I didn't spend too much time thinking about it.

Reply:
I can see it being fine, I dont think most people are going to touch it. However I closed my PR here and PRed this branch with my convenience function that is the more common representation just checking ZYXDA. The idea is if you're polling the sensor for data as fast as possible, you need to actually wait until it updates ZYXDA before you try to read fresh data.

Reply:
Yeah, doc comments with the explanation you just provided would be sufficient.

Reply:
So if you're only reading x, you only need to wait til x is ready. 

But our common use case is raw_accell grabbing all three so code in a tight loop SHOULD be doing
```
            while !lis3dh.is_data_ready().unwrap() {}
            let dat = lis3dh.accel_raw().unwrap();
```


My pr waiting for @jessebraham  to approve is here
https://github.com/jessebraham/lis3dh-rs/pull/1

Reply:
Oh and overflow means that two reading wrote over eachother before you read were able to read. Most people dont care, but I could see adding an `is_overflowed()` or something in the future which would be just ZYXOR

Reply:
@jacobrosenthal thanks for the contribution.
I've added the doc comment.

Reply:
Any reason not to bake the call to `is_data_ready` into `accel_raw`?

Reply:
Its not a terrible Idea, I just dont think Ive seen anyone else do it so im not sure its idiomatic for accel_raw...

Well I searched github, theres actually one guy doing it
https://github.com/JosephBGerber/lis3dsh-rs/blob/5ef60fa5d95199df18fcfb20d6ea1a2e810c36b7/src/lib.rs

It adds an extra round trip which might annoy some zero costers. And people might be confused and not understand to check for a WouldBlock thats not actually an error...

Might be appropriate to bring up in an issue in the accelerometer trait crate.

Reply:
https://github.com/NeoBirth/accelerometer.rs/issues/61

Reply:
I haven't seen other crates doing that. My gut is leaning towards maintaining the separation of concerns here, but I don't feel too strongly either way about it.